src/Orleans.Runtime/GrainTypeManager/GrainTypeManager.cs

```diff
@@ -34,19 +34,42 @@ internal class GrainTypeManager
 
         public IGrainTypeResolver GrainTypeResolver { get; private set; }
 
+        private readonly ILogger<GrainInterfaceMap> loggerGrainInterfaceMap;
+        private readonly Advanced.IGrainTypeResolver grainTypeResolver;
+        private readonly Advanced.IInterfaceTypeResolver interfaceTypeResolver;
+
         public GrainTypeManager(
             ILocalSiloDetails siloDetails,
             IApplicationPartManager applicationPartManager,
             PlacementStrategy defaultPlacementStrategy,
             SerializationManager serializationManager,
+            ILogger<GrainInterfaceMap> loggerGrainInterfaceMap,
             ILogger<GrainTypeManager> logger,
             IOptions<GrainClassOptions> grainClassOptions)
+            : this(siloDetails, applicationPartManager, defaultPlacementStrategy, serializationManager, loggerGrainInterfaceMap, logger, grainClassOptions, null, null)
+        {
+        }
+
+        public GrainTypeManager(
```

Comment:
Did you duplicate the constructor to keep usage of the old one? If I am not mistaken, this type is only constructed by DI, and I would rather not rely on the DI logic to select a constructor in this case, if that make sense

Reply:
for non-migration scenarios, it will choose a constructor without extra interface parameters, because they are not registered. Do you want me to rewrite to a explicit resolve of components? I can do that to not rely on DI (IMO its OK as it is now)

Reply:
addressed this